packages/node-core/src/indexer/indexer.manager.ts

```diff
@@ -116,6 +116,32 @@ export abstract class BaseIndexerManager<
           dynamicDsCreated = true;
         }, 'createDynamicDatasource');
 
+        // Inject function to get dynamic datasources by template into vm
+        vm.freeze((templateName: string) => {
+          return this.dynamicDsService.getDynamicDatasourcesByTemplate(templateName);
+        }, 'getDynamicDatasources');
+
+        // Inject function to destroy ds into vm
+        vm.freeze(async (templateName: string, index: number) => {
+          await this.dynamicDsService.destroyDynamicDatasource(templateName, blockHeight, index);
+
+          // Remove destroyed datasources from current processing
+          // Filter out datasources that have been destroyed
+          const updatedFilteredDataSources = filteredDataSources.filter((fds) => {
+            const dsParam = (this.dynamicDsService as any)._datasourceParams?.find(
+              (p: any) =>
+                p.templateName === (fds as any).mapping?.file?.split('/').pop()?.replace('.js', '') ||
+                (p.startBlock === (fds as any).startBlock && p.templateName === templateName)
```

Comment:
I don't think this is sufficient, there could be multiple instances of a dynamic datasource with the same startBlock but different args.